src/vs/platform/files/browser/htmlFileSystemProvider.ts

```diff
@@ -365,7 +365,7 @@ export class HTMLFileSystemProvider implements IFileSystemProviderWithFileReadWr
 		}
 	}
 
-	private async getDirectoryHandle(resource: URI): Promise<FileSystemDirectoryHandle | undefined> {
+	async getDirectoryHandle(resource: URI): Promise<FileSystemDirectoryHandle | undefined> {
```

Comment:
cc @bpasero we need the raw `FileSystemDirectoryHandle` on the webworker side to ensure minimum possible overhead when crawling through the entire directory.

Reply:
Not really having a better suggestion, but I would rather make `getHandle` public and then check on the returned type to be a directory. 

The entire way how to get handles in and out of the provider is a bit of a hack, as indicated by how the file dialog service uses it today:

https://github.com/microsoft/vscode/blob/17314f56bab57d945fb16cf30ca84b8539afadb7/src/vs/workbench/services/dialogs/browser/fileDialogService.ts#L24-L26

Reply:
Sounds good, changed is so `getHandle` is now the public one. Saw that file service snippet, it was the basis for what I did here. 

It'd be cool to have the handles stored for reload, I wonder how the browsers would handle it. I was a bit surprised to see I could send the handle over to a webworker so easily.

Reply:
I think that is really the idea that you can persist and transfer these handles to use them in other contexts or later. However, you will have to call `queryPermission`: https://web.dev/file-system-access/#stored-file-or-directory-handles-and-permissions